content/docs/rfcs/56/README.md

```diff
@@ -0,0 +1,412 @@
+
+### Advertising a Community
+
+The content topic that the community is advertised on MUST be derived from the public key of the community.
+The content topic MUST be the hex-encoded keccak-256 hash of the compressed (32 bytes) public key of the community.
+
+```
+contentTopic = hex(keccak256(compressedPublicKey))
+```
+
+### Community channels/chats
+
+The unique identifier for a community channel/chat is the chat id.
+<!-- Don't enforce any constraints on the unique id generation -->
+The content topic that Community channels/chats use MUST be the hex-encoded keccak-256 hash of the public key of the community concatenated with the chat id.
+
+```
+contentTopic = hex(keccak256(compressedPublicKey + chatId))
+```
+
+
+### Community event messages
+
+Requests to leave, join, kick and ban, as well as key exchange messages, MUST be sent to the content topic derived from the public key of the community.
+The content topic MUST be the hex-encoded keccak-256 hash of the public key of the community.
+
+```
+contentTopic = hex(keccak256(publicKey))
+``` 
+
+## Community Management
+
+The flows for Community management are as described below.
+
+### Community Creation Flow
+
+1. The Community owner generates a public/private key pair.
+2. The Community owner configures the Community metadata, according to the wire format "CommunityDescription".
+3. The Community owner publishes the Community metadata on a content topic derived from the public key of the Community. 
+the Community metadata SHOULD be encrypted with the public key of the Community. <!-- TODO: Verify this-->
+The Community metadata MAY be sent during fixed intervals, to ensure that the Community metadata is available to members.
+The Community metadata SHOULD be sent every time the Community metadata is updated.
+4. The Community owner MAY advertise the Community out of band, by sharing the public key of the Community on other mediums of communication.
+
+### Community Join Flow (peer requests to join a Community)
+
+1. A peer generates a public/private key pair.
+2. The peer requests to join a Community by sending a "CommunityRequestToJoin" message to the Community.
+At this point, the peer MAY send a "CommunityCancelRequestToJoin" message to cancel the request.
+3. The Community owner MAY accept or reject the request.
+4. If the request is accepted, the Community owner sends a "CommunityRequestToJoinResponse" message to the peer.
+5. The Community owner then adds the member to the Community metadata, and publishes the updated Community metadata.
+
+### Community Join Flow (peer is invited to join a Community)
+
+1. The peer is invited to join a Community by the Community owner, by sending a "CommunityInvitation" message.
+2. A peer generates a public/private key pair.
+3. The peer decrypts the "CommunityInvitation" message, and verifies the signature.
```

Comment:
back to my comment above. How does the peer decrypt the message? I guess not with the key it just generated.

Reply:
Made it more clear in [3690bab](https://github.com/vacp2p/rfc/pull/567/commits/3690baba8f84cbe08393a7e1c1a514ab7f2e7683)